### PR TITLE
Added Vietnamese translation

### PR DESCRIPTION
**Pull request by MenacingExiler**
Translated to Vietnamese. Note this is untested and fully human translated. Some words make little sense in direct translation so I replaced with more familiar alternatives. There will be more improvements in future.

### CODE REVIEW

File: gui/src/i18n/config.ts [hunk 1]
```diff
@@ -3,8 +3,9 @@ import { initReactI18next } from 'react-i18next';
 import en from './en/translation.json';
 import fr from './fr/translation.json';
 import owo from './owo/translation.json';
+import vi from "./vi/translation.json"
```
Comment (Louka3000) [inline gui/src/i18n/config.ts:6]:
```suggestion
import vi from './vi/translation.json';
```

**Review by Louka3000 — CHANGES_REQUESTED** [review]
Could you move vietnamese in config above owo furry language?
I think joke languages should be at the very bottom (unless we want it to be right below English, but it'd still need changing even then)

File: gui/src/i18n/config.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@ import { initReactI18next } from 'react-i18next';
...
 
-export const defaultNS = 'tanslations';
+export const defaultNS = 'translations';
```
Comment (Louka3000) [inline gui/src/i18n/config.ts:9]:
```suggestion
export const defaultNS = 'translation';
```

File: gui/src/i18n/vi/translation.json [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+      "tracker-mechanics": "Cơ khí tracker",
+      "fk-settings": "Cài đặt FK",
+      "interface": "Giao diện",
+      "utils": "Hữu dụng",
+      "serial": "Serial Console"
+    },
+    "general": {
+      "steamvr": {
+        "title": "SteamVR",
+        "subtitle": "SteamVR trackers",
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:171]: This wasn't translated

File: gui/src/i18n/vi/translation.json [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+    "full": "Reset",
+    "mounting": "Reset vị trí cài",
+    "quick": "Reset nhanh"
+  },
+  "serial-detection": {
+    "new-device.p0": "New serial device detected!" ,
+    "new-device.p1": "Vui lòng nhập thông tin Wi-Fi!",
+    "new-device.p2": "Vui lòng chọn những gì làm với nó",
+    "open-wifi": "Kết nối Wi-Fi",
+    "open-serial": "Mở Serial Console",
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:65]: same here?

File: gui/src/i18n/vi/translation.json [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+  },
+  "settings": {
+    "sidebar": {
+      "title": "Cài đặt",
+      "general": "Cài đặt chung",
+      "tracker-mechanics": "Cơ khí tracker",
+      "fk-settings": "Cài đặt FK",
+      "interface": "Giao diện",
+      "utils": "Hữu dụng",
+      "serial": "Serial Console"
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:166]: forgot translation

File: gui/src/i18n/vi/translation.json [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+    "new-device.p0": "New serial device detected!" ,
+    "new-device.p1": "Vui lòng nhập thông tin Wi-Fi!",
+    "new-device.p2": "Vui lòng chọn những gì làm với nó",
+    "open-wifi": "Kết nối Wi-Fi",
+    "open-serial": "Mở Serial Console",
+    "submit": "Đăng lên!",
+    "close": "Đóng"
+  },
+  "navbar": {
+    "home": "Home",
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:70]: this too

File: gui/src/i18n/vi/translation.json [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+      "disconnected": "Đã ngắt kết nối",
+      "occluded": "Bị tắc",
+      "ok": "Đã kết nối"
+    },
+    "table": {
+      "column": {
+        "name": "Tên",
+        "type": "Loại",
+        "battery": "Pin",
+        "ping": "Ping",
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:99]: this is latency fyi.

**Comment by MenacingExiler** [comment]
Got it I changed order and forgot to note I also fixed typos

File: gui/src/i18n/vi/translation.json [hunk 8]
```diff
@@ -0,0 +1,460 @@
...
+      },
+      "mounting-section": {
+        "title": "Vị trí đặt bộ phận",
+        "description": "Tracker đặt ở đâu?",
+        "edit": "Chỉnh lại chỗ đặt"
+      },
+      "name-section": {
+        "title": "Tên tracker",
+        "description": "Hãy cho một tên đẹp :)",
+        "input-placeholder": "NightyBeast's left leg"
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:133]: Change this to something funny :3
You can do "Menaced's left leg" if you want!

File: gui/src/i18n/vi/translation.json [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+      "front": "Đằng trước",
+      "left": "Bên trái",
+      "right": "Bên phải",
+      "back": "Đằng sau" 
+    },
+    "infos": {
+      "manufacturer": "Nhà sản xuất",
+      "display-name": "Tên gọi",
+      "custom-name": "Tên tự chọn",
+      "url": "Tacker URL"
```
Comment (Louka3000) [inline gui/src/i18n/vi/translation.json:115]: Tacker should be Tracker
Also is there not a translation for "Tracker"? It seems you didn't translate it anywhere.

**Review by Louka3000 — CHANGES_REQUESTED** [review]
Apart from the missing translation and other error in config, it looks good (altho I can't review the actual translations).
Feels awesome to have the GUI translated in other languages!

File: gui/src/i18n/vi/translation.json (hunk 9 above)
Comment (MenacingExiler) [reply gui/src/i18n/vi/translation.json:115]: Right, about that I could translate tracker but Vietnamese especially young would be more associated with word tracker.

**Comment by MenacingExiler** [comment]
Fixed all issues commented. There are some untranslated word because I believe Vietnamese are more associated with foreign lent words than Vietnamese word of it and I do not feel like using complicated Han-Vietnamese lend words to make it too academic or formal rather I translated more modern and Vietnamese-English mixed commonly familiar nowadays. However, if requested I can change translation style on request. Personally, I would prefer getting feedback after release to see if people want it different but for now I want to avoid corporate style but not too familiar way meanings.

**Comment by MenacingExiler** [comment]
> Apart from the missing translation and other error in config, it looks good (altho I can't review the actual translations). Also fix this like I mentioned in Discord: 
> 
> Feels awesome to have the GUI translated in other languages!

Fixed all issues commented. There are some untranslated word because I believe Vietnamese are more associated with foreign lent words than Vietnamese word of it and I do not feel like using complicated Han-Vietnamese lend words to make it too academic or formal rather I translated more modern and Vietnamese-English mixed commonly familiar nowadays. However, if requested I can change translation style on request. Personally, I would prefer getting feedback after release to see if people want it different but for now I want to avoid corporate style but not too familiar way meanings.

File: gui/src/i18n/config.ts [hunk 10]
```diff
@@ -2,9 +2,11 @@ import i18next from 'i18next';
 import { initReactI18next } from 'react-i18next';
 import en from './en/translation.json';
 import fr from './fr/translation.json';
-import owo from './owo/translation.json';
+import vi from './vi/translation.json';
+import owo from './owo/translation.json'
```
Comment (carl-anders) [inline gui/src/i18n/config.ts:6]: Missing ; at the end?

**Comment by carl-anders** [comment]
> Fixed all issues commented. There are some untranslated word because I believe Vietnamese are more associated with foreign lent words than Vietnamese word of it and I do not feel like using complicated Han-Vietnamese lend words to make it too academic or formal rather I translated more modern and Vietnamese-English mixed commonly familiar nowadays. However, if requested I can change translation style on request. Personally, I would prefer getting feedback after release to see if people want it different but for now I want to avoid corporate style but not too familiar way meanings.

I agree completely with using mixed languages when it is relevant for the language. It sounds like Vietnamese is similar to Swedish in that way, where keeping important loanwords is needed for software translations. One of the reasons using software translated to Swedish is often unusable is because of forced translations to concepts that doesn't actually map correctly.

The word "**tracker**" is a lot more readable than **"hunter-searching-for-pray"**, or whatever a native translation would involve.

**Comment by Louka3000** [comment]
Yes it’s fine ofc ^^

**Comment by Louka3000** [comment]
Tested, it works. Good for merge :3

**Comment by Eirenliel** [comment]
> Tested, it works. Good for merge :3

then approve it :nya_gun:

**Review by Louka3000 — APPROVED** [review]
Forgor to approve

**Pull request merged**